### PR TITLE
feat(atomic): make grid result <a> a sibling instead of parent

### PR DESCRIPTION
**Pull request by ThibodeauJF**
https://coveord.atlassian.net/browse/KIT-1574

Why? https://html.spec.whatwg.org/#the-a-element

> [Transparent](https://html.spec.whatwg.org/#transparent), but there must be no [interactive content](https://html.spec.whatwg.org/#interactive-content-2) descend[a](https://html.spec.whatwg.org/#the-a-element)nt, a element descendant, or descendant with the [tabindex](https://html.spec.whatwg.org/#attr-tabindex) attribute specified.

Inspired by https://css-tricks.com/nested-links/ & https://inclusive-components.design/cards/

### CODE REVIEW

File: packages/atomic/src/components/result-lists/result-list-common.pcss [hunk 1]
```diff
@@ -73,9 +73,18 @@
...
+  cursor: pointer;
+}
+
+[part='result-list-grid-clickable'] {
+  position: absolute;
+  top: 0;
+  bottom: 0;
+  right: 0;
+  left: 0;
+  pointer-events: none;
```
Comment (liviarett) [inline packages/atomic/src/components/result-lists/result-list-common.pcss:87]: doesn't this prevent the click event from being dispatched?

Comment (ThibodeauJF) [reply packages/atomic/src/components/result-lists/result-list-common.pcss:87]: 🤦 🤦 🤦 🤦 🤦

File: packages/atomic/src/components/result-lists/grid-display-results.tsx [hunk 2]
```diff
@@ -12,14 +12,16 @@ export const GridDisplayResults: FunctionalComponent<ResultsProps> = (
...
     return (
-      <LinkWithResultAnalytics
-        part="result-list-grid-clickable"
-        onSelect={() => interactiveResult.select()}
-        onBeginDelayedSelect={() => interactiveResult.beginDelayedSelect()}
-        onCancelPendingSelect={() => interactiveResult.cancelPendingSelect()}
-        href={props.resultListCommon.getUnfoldedResult(result).clickUri}
-        target="_self"
-      >
+      <div part="result-list-grid-parent">
```
Comment (olamothe) [inline packages/atomic/src/components/result-lists/grid-display-results.tsx:15]: This part does not seem properly documented

File: packages/atomic/src/components/result-lists/result-list-common.pcss (hunk 1 above)
Comment (ThibodeauJF) [reply packages/atomic/src/components/result-lists/result-list-common.pcss]: Reworking lol

**Review by olamothe — APPROVED** [review]

**Comment by ThibodeauJF** [comment]
To disable would just be 

```
atomic-result-list::part(result-list-grid-clickable) {
  display: none;
}
```

File: packages/atomic/src/components/result-lists/grid-display-results.tsx (hunk 2 above)
Comment (olamothe) [inline packages/atomic/src/components/result-lists/grid-display-results.tsx:15]: Suggestion: 
```suggestion
      <div part="result-list-grid-clickable-container">
```

**Review by olamothe — APPROVED** [review]

File: packages/atomic/src/components/result-lists/grid-display-results.tsx (hunk 2 above)
Comment (ThibodeauJF) [reply packages/atomic/src/components/result-lists/grid-display-results.tsx:15]: hmm it contains the clickable link & the result too? maybe `result-list-grid-container`

**Review by liviarett — APPROVED** [review]
Looks good, but looking at the description you added made me wonder if a screen reader wouldn't consider this a clickable link like any other... and then if you actually select the result with it, it won't send the analytics? I don't know if that makes sense, but just a thought!

**Comment by ThibodeauJF** [comment]
Good question!

Yes it is a link taking the whole space, if you click anywhere (except a child a or button) it redirects you & logs a click. Other links that we have such as result link already log analytics so it's good. In non-grid results, that's already how it works

**Review by jpmarceau — APPROVED** [review]
👍

**Pull request merged**